go_to_office/lib/screens/login.dart

```diff
@@ -19,52 +20,111 @@ class _LoginPageState extends State<LoginPage> {
   _LoginPageState(this.repository);
   final Repository repository;
 
+  void onUserLoggedIn() {
+    Navigator.push(
+      context,
+      MaterialPageRoute(
+          builder: (context) => Meetings(
+            title: "Meetings",
+          )),
+    );
+  }
+
+  void onAdminLoggedIn() {
+    Navigator.push(
+      context,
+      MaterialPageRoute(builder: (context) => OfficesListPage(),
+    ));
+  }
+
   final _userTextController = TextEditingController();
   final _passTextController = TextEditingController();
+  final GlobalKey<ScaffoldState> _scaffoldKey = new GlobalKey<ScaffoldState>();
 
   @override
   void dispose() {
-    // Clean up the controller when the widget is disposed.
+    // Cleans up the controller when the widget is disposed.
     _userTextController.dispose();
     _passTextController.dispose();
     super.dispose();
   }
 
   @override
   Widget build(BuildContext context) {
-    void showSnack(String text) {
-      final snackBar = SnackBar(content: Text(text));
-      // TBC -> implement it by Oz
+
+    void _showSnack(String text) {
+      _scaffoldKey.currentState.showSnackBar( SnackBar(content: Text(text)) );
+    }
+
+    void _showError(String text) {
+      _scaffoldKey.currentState.showSnackBar( SnackBar(content: Text(text), backgroundColor: Colors.red) );
+    }
+
+    void _onError(Object error) {
+      String text = "";
+      if (error is FirebaseAuthException) {
+        text = error.message;
+      } else {
+        text = error.toString();
+      }
+      _showError(text);
     }
 
-    void onError(String text) {
-      final snackBar =
-          SnackBar(content: Text(text), backgroundColor: Colors.red);
-      // TBC -> implement it by Oz
+    bool _userCredentialsValidation(String _email, String _password) {
+
+      bool _validPassword = _password.isNotEmpty && _password.length > 5;
```

Comment:
you need to write _password?.isNotEmpty for coving a situation where the _password is null
same goes for the _email of course.